### PR TITLE
Con-2149

### PR DESCRIPTION
**Pull request by bhagyashree-sarawate**
@raunakkumar 
Following attributes should be configurable : 
CSP - labels, tolerations, affinity, node selectors

Controller - labels, tolerations, affinity, node selectors

CSI Node - labels, tolerations, affinity, node 

feature page link 
https://confluence.eng.nimblestorage.com/display/ESI/CON-2149

### CODE REVIEW

**Review by datamattsson — CHANGES_REQUESTED** [review]

File: operators/hpe-csi-operator/deploy/crds/storage.hpe.com_v1_hpecsidriver_cr.yaml [hunk 1]
```diff
@@ -38,3 +38,96 @@ spec:
   # NodeGetVolumestats will be called by default, set true to disable the call
   disableNodeGetVolumeStats: false
 
+  controller:
+    # These values map directly to yaml in the deployment spec, see the kubernetes docs for info
+    labels: { }
```
[the remark this answers is not in the document]
Comment (bhagyashree-sarawate) [reply operators/hpe-csi-operator/deploy/crds/storage.hpe.com_v1_hpecsidriver_cr.yaml]: done

[the remark this answers is not in the document]
Comment (bhagyashree-sarawate) [reply operators/hpe-csi-operator/deploy/crds/storage.hpe.com_v1_hpecsidriver_cr.yaml]: @raunakkumar @datamattsson can you please review this PR?

**Review by raunakkumar — APPROVED** [review]

File: operators/hpe-csi-operator/deploy/crds/storage.hpe.com_v1_hpecsidriver_cr.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (bhagyashree-sarawate) [reply operators/hpe-csi-operator/deploy/crds/storage.hpe.com_v1_hpecsidriver_cr.yaml]: @datamattsson , i have addressed review comments. Let me know if anything is pending from my side

**Review by datamattsson — APPROVED** [review]

**Comment by bhagyashree-sarawate** [comment]
@Raunak, @datamattsson , when we can merge this, should I add the pre-release tag?

**Comment by raunakkumar** [comment]
> @raunak, @datamattsson , when we can merge this, should I add the pre-release tag?

I have merged it. Once it has gone through a round of internal testing we can mark with a pre-release tag

**Pull request merged**